### PR TITLE
Provide a default component to use for managing row edit mode

### PR DESCRIPTION
**Pull request by bgantzler**
I believe this satisfies all the requirements.

I updated the example to use the provided one instead and deleted the one in the example.

You use double quotes in the hbs, and java, javascript uses both. So I just use double quotes as its accepted by all three. Putting the lint that says single quotes only caused me a lot of typos :)

### CODE REVIEW

File: addon/components/models-table/cell-edit-toggle.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  cancelRowAction: null,
+
+  /**
+   * The label for the Edit Button
+   *
+   * @property editLabel
+   * @type string
+   * @default themeInstance.labels.editRowButton
+   */
+  editLabel: computed({
```
Comment (onechiporenko) [inline addon/components/models-table/cell-edit-toggle.js]: Do we need a default setter?

File: addon/components/models-table/cell-edit-toggle.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  }),
+
+  /**
+   * The label for the Cancel Button
+   *
+   * @property cancelLabel
+   * @type string
+   * @default themeInstance.labels.cancelRowButton
+   */
+  cancelLabel:computed({
```
Comment (onechiporenko) [inline addon/components/models-table/cell-edit-toggle.js]: Do we need a default setter?

File: addon/components/models-table/cell-edit-toggle.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  }),
+
+  /**
+   * The label for the Save Button
+   *
+   * @property editLabel
+   * @type string
+   * @default themeInstance.labels.saveRowButton
+   */
+  saveLabel: computed({
```
Comment (onechiporenko) [inline addon/components/models-table/cell-edit-toggle.js]: Do we need a default setter?

File: addon/themes/default.js [hunk 4]
```diff
@@ -54,6 +58,12 @@ export default EmberObject.extend({
     'cell-content': ''
   },
 
+  labels: {
```
Comment (onechiporenko) [inline addon/themes/default.js]: Can we call this `messages` (I have a plan to move `messages` from `models-table` to `theme`)?

File: addon/components/models-table/cell-edit-toggle.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  /**
+   * Closure action sent on Save Button being clicked
+   *
+   * Action will send one parameter - object with fields:
+   *
+   * * `record` - The record to be saved
+   *
+   * @type function
+   * @property saveRowAction
+   * @default null
```
Comment (onechiporenko) [inline addon/components/models-table/cell-edit-toggle.js]: Bound closure actions are marked as `@event` and not `@property` with `@type` and `default`

File: addon/components/models-table/cell-edit-toggle.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  /**
+   * Closure action sent on Cancel Button being clicked
+   *
+   * Action will send one parameter - object with fields:
+   *
+   * * `record` - The record currently being edited
+   *
+   * @type function
+   * @property cancelRowAction
+   * @default null
```
Comment (onechiporenko) [inline addon/components/models-table/cell-edit-toggle.js]: Bound closure actions are marked as `@event` and not `@property` with `@type` and `default`

File: addon/components/models-table/cell-edit-toggle.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+  /**
+   * Closure action sent on Edit Button being clicked
+   *
+   * Action will send one parameter - object with fields:
+   *
+   * * `record` - The record to be edited
+   *
+   * @type function
+   * @property editRowAction
+   * @default null
```
Comment (onechiporenko) [inline addon/components/models-table/cell-edit-toggle.js]: Bound closure actions are marked as `@event` and not `@property` with `@type` and `default`

**Comment by onechiporenko** [comment]
@bgantzler, thanks for PR! It looks realy good! I left a few comments. Some of them require changes and some are discussable.

File: addon/components/models-table/cell-edit-toggle.js (hunk 1 above)
Comment (bgantzler) [reply addon/components/models-table/cell-edit-toggle.js]: The setters allow for 
```
{{models-table
  data=data
  columns=columns
  columnComponents=(hash
    editRow=(component "models-table/cell-edit-toggle" saveRowAction=(action "onSaveRow") cancelRowAction=(action "onCancelRow")) editLabel="MyEditLabel" cancelLabel="MyCancelLabel"
  )
}}

```

File: addon/themes/default.js (hunk 4 above)
Comment (bgantzler) [reply addon/themes/default.js]: Would you want the word Label added to the elements to better call out the purpose
```
  messages: {
    editRowButtonLabel: 'Edit',
    saveRowButtonLabel: 'Save',
    cancelRowButtonLabel: 'Cancel'
  },

```

Comment (onechiporenko) [reply addon/themes/default.js]: Yes, this would be great.

File: addon/components/models-table/cell-edit-toggle.js (hunk 1 above)
Comment (onechiporenko) [reply addon/components/models-table/cell-edit-toggle.js]: Ok. Let's leave it as is.

**Comment by bgantzler** [comment]
I have made the requested changes and re-pushed the request. Not sure how to mark on the request that this was completed

**Comment by onechiporenko** [comment]
Thanks, @bgantzler!

**Pull request merged**